libs/feature/search/src/lib/utils/service/fields.ts

```diff
@@ -220,13 +234,78 @@ if (doc.containsKey('licenseObject.default.keyword') && doc['licenseObject.defau
 if (doc.containsKey('MD_LegalConstraintsUseLimitationObject.default.keyword') && doc['MD_LegalConstraintsUseLimitationObject.default.keyword'].length > 0)
   raw += doc['MD_LegalConstraintsUseLimitationObject.default.keyword'].join('|').toLowerCase();
 
-if (raw.contains('odbl')) emit('ODbL');
-if (raw.contains('pddl')) emit('PDDL');
-if (raw.contains('odc-by')) emit('ODC-By');
-if (raw.contains('cc0') || raw.contains('cc-0')) emit('CC-0');
-if (raw.contains('cc-by') || raw.contains('cc by')) emit('CC BY');
-if (raw.contains('open license')) emit('Open License');
-if (raw.contains('etalab')) emit('Etalab');`
+boolean unknown = true;
+if (raw.contains('pddl') || raw.contains('public domain dedication and licence')) {
```

Comment:
I also think that this is a code related to licence mapping and should be agnostic to this service.
For instance, if I implement harvesting with geonetwork-ui, I might need the same logic while I wouldn't need the `FieldService` things.
Let's see.

Reply:
In theory yes... but this is the painless language which is ES specific, I'm not sure how we could factorize that. I considered having an object describing for each licence a set of keywords to look for and generate the painless script out of it, but then it would be too complex I think.